src/libNetwork/P2PComm.cpp

```diff
@@ -394,15 +394,8 @@ void P2PComm::ClearBroadcastHashAsync(const vector<unsigned char>& message_hash)
 void P2PComm::EventCallback(struct bufferevent* bev, short events,
                             [[gnu::unused]] void* ctx)
 {
-    // Get the IP info
-    int fd = bufferevent_getfd(bev);
-    struct sockaddr_in cli_addr;
-    socklen_t addr_size = sizeof(struct sockaddr_in);
-    getpeername(fd, (struct sockaddr*)&cli_addr, &addr_size);
-    Peer from(cli_addr.sin_addr.s_addr, cli_addr.sin_port);
-
-    // Close the socket
-    bufferevent_free(bev);
+    unique_ptr<struct bufferevent, void (*)(struct bufferevent*)> socket_closer(
```

Comment:
for simplicity and better connection with the free function, we could use `decltype(bufferevent_free)` here